Lib/test/test_pyrepl/test_pyrepl.py

```diff
@@ -1100,6 +1100,31 @@ def test_not_wiping_history_file(self):
         self.assertIn("spam", output)
         self.assertNotEqual(pathlib.Path(hfile.name).stat().st_size, 0)
 
+    @force_not_colorized
+    def test_proper_tracebacklimit(self):
+        env = os.environ.copy()
+        commands = ("import sys\n"
+                    "sys.tracebacklimit = 1\n"
+                    "def x1(): 1/0\n\n"
+                    "def x2(): x1()\n\n"
+                    "def x3(): x2()\n\n"
+                    "x3()\n"
+                    "exit()\n")
+
+        env.pop("PYTHON_BASIC_REPL", None)
+        output, exit_code = self.run_repl(commands, env=env)
+        if "can\'t use pyrepl" in output:
+            self.skipTest("pyrepl not available")
+        self.assertIn("in x1", output)
+        self.assertNotIn("in x3", output)
```

Comment:
What about "not in x2"?

Negative tests are tricky. How can we sure that it is passed because of tracebacklimit and not because of a typo? `self.assertNotIn("abrakadabra", output)` also passed.

Is there a test that tests the same code, but without setting tracebacklimit? If not, we can do it here.

Reply:
I've added checks for `x2`. I've also added a case for *not* setting the tracebacklimit, to check that without a limit the "not in" strings are indeed in the output